[issue_tracker] Update testing plan for issue attachments (#6217)
aces/Loris: modules/issue_tracker/test/issue_tracker_test_plan.md

Issue Tracker Filter Form [Automation Testing]

> 1. User can access the page if they have issue tracker reporter or developer permission

Reviewer: ```suggestion
1. A user can access the module as long as they have one of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.
```

iff is a short form for "if, and only if,". I agree that it doesn't provide much clarity in this context though so I'm happy to reword the sentence. 

Item 2 should also be reworded.

Author: Wow, I didn't know. Thanks @johnsaigle

> 1. A user can access the module as long as they have one of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.

Reviewer 2: ```suggestion
1. A user can access the module only if they have one or more of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.
```

4. Submit invalid and valid PSCID and visit label pairs. Error messages should respond accordingly. Not that you cannot submit PSCIDs from other sites unless you have access all profiles permission
5. Submit just a visit label - this should give an error message.
6. Check that all values are propagated and saved correctly.

> 7. Submit with an uploaded attachment should result in success.

Reviewer: ```suggestion
7. Add an attachment to the new issue and make sure that it is successfully uploaded.
```

5. Submit just a visit label - this should give an error message.
6. Check that all values are propagated and saved correctly.
7. Check that watching logging is working - turn it off and on for your current user, and for other watchers on the issue

> 8. Check uploading new attachment to existing issue.

Reviewer: ```suggestion
8. Check that an attachment can be adding to an existing issue.
```

Permissions [Automation Testing]
1. Remove access all profile permission.
2. Remove reporter permission
3. Remove developer permission
4. Test that the module behaves correctly as described above. 

> 5. Test if user assigned to issue cannot delete attachments of issue owner.

Reviewer: Have steps 5-7 been automated? If not they should not be under this heading of "Automation Testing"

Author: @johnsaigle oops mistake.

4. Test that the module behaves correctly as described above. 

> 5. Test if user assigned to issue cannot delete attachments of issue owner.

Reviewer: Is this automated?

Author: thanks @johnsaigle, I somehow didn't remove that.
